### PR TITLE
Allows admins to interact with most machines like an AI [Review Ready]

### PR DESCRIPTION
**Pull request by Shadowlight213**
Some objects excluded for being able to have their functions easily reproduced with admin verbs.

Still has a bug with NanoUI where it is giving script errors. DNM due to that.
@neersighted going to need your help with solving the nanoui issue.

:cl: Shadowlight213
rscadd: Admins can now interact with mos of the station's machinery as if they were an AI when ghosting.
/:cl:

### CODE REVIEW

**Comment by WJohn** [comment]
:+1: x :100:

**Comment by AlexCrimson** [comment]
I hope this is logged. Otherwise i cannot trust admins to not delete my saved data whilst doing Genetics or R&D.

**Comment by Shadowlight213** [comment]
This is logged at the same level that an AI doing it is logged. I'll add improved logging once I can get the errors solved.

**Comment by AnturK** [comment]
IsAGhost is bit too similar to IsGhost. I'd change that to IsAdminGhost or something similar.

**Comment by Shadowlight213** [comment]
K

**Comment by Shadowlight213** [comment]
Ok. Bug seems to be finally fixed. Ready for review.

File: code/game/objects/objs.dm [hunk 1]
```diff
@@ -93,9 +93,9 @@
 
 
 /obj/attack_ghost(mob/user)
+	if(ui_interact(user) != -1)
```
Comment (Shadowlight213) [inline code/game/objects/objs.dm]: This returns -1 if What was clicked on wasn't a Nanoui. Basically this prevents both the attack_ai proc and the interact proc being called at the same time.

**Comment by Shadowlight213** [comment]
@Jordie0608 @duncathan @RemieRichards 
Need review please.

File: code/game/machinery/computer/medical.dm [hunk 2]
```diff
@@ -248,6 +248,12 @@
 				src.authenticated = 1
 				src.rank = "AI"
 				src.screen = 1
+			else if(IsAdminGhost(usr))
+				src.active1 = null
+				src.active2 = null
+				src.authenticated = 1
+				src.rank = "Central Command"
```
Comment (Niknakflak) [inline code/game/machinery/computer/medical.dm]: Hilarious reference here because normally these guys have zero access

**Pull request merged**